src/library/dao/playlistdao.cpp

```diff
@@ -380,13 +380,96 @@ bool PlaylistDAO::isHidden(const int playlistId) const {
     return true;
 }
 
-void PlaylistDAO::removeTrackFromPlaylist(const int playlistId, const int position)
-{
+
+void PlaylistDAO::removeHiddenTracks(const int playlistId) {
+	ScopedTransaction transaction(m_database);
+    // This query deletes all tracks marked as deleted and all
+    // phantom track_ids with no match in the library table
+    QString queryString = QString(
+            "SELECT position FROM PlaylistTracks "
```

Comment:
Doing it like this was the easiest way for me because I could reuse existing code, without much testing. My original idea was just to fix the issue of AutoDJ and postpone the rest ... now it is even better but not perfect. 

I have not thought much about a good way to close all gaps and find duplicates in an effective way. 
I can only think of a query, sort by number and compare number with index. If It does not match, fix it. Such a code would be required to fix issues in old libraries. 

Is it OK to postpone this to an other PR?